src/c++/perf_analyzer/infer_context.cc

```diff
@@ -113,7 +113,8 @@ InferContext::SendRequest(
 
   thread_stat_->num_sent_requests_++;
   if (async_) {
-    infer_data_.options_->request_id_ = std::to_string(request_id);
+    uint64_t unique_request_id{(thread_id_ << 48) | ((request_id << 16) >> 16)};
```

Comment:
Did we get any indication that using a string won't work? This would be a lot harder to parse than just string-concatenating the two values.  When someone sees a request-ID of 281 trillion they might be confused or expect that it is a bug.

Reply:
https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/inflight_batcher_llm/src/work_item.h#L71

This indicates the TensorRT-LLM Triton backend expects integer.

@fpetrini15 maybe some comment on this? Or maybe someone else (feel free to tag)?

Reply:
https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/inflight_batcher_llm/src/work_item.cc#L34

This is where a random integer is auto-generated when request ID is not provided.

Reply:
@matthewkotila what you said is correct. The TRT LLM backend expects a uint64 and appears to be tightly coupled in the design of TRT LLM (ex. inference request [get id](https://github.com/NVIDIA/TensorRT-LLM/blob/71f60f6df09b29ab33861f98ac8d97cbe5417e45/cpp/include/tensorrt_llm/batch_manager/inferenceRequest.h#L107)). That said, if there is way to make your lives easier here with an additional method lmk.